app/validation/validator.py

```diff
@@ -742,16 +742,23 @@ def _validate_routing_when_dict(self, when_dict, answer_ids_with_parent_id):
         errors = []
         when_values = when_dict.get("values", [])
         when_value = when_dict.get("value")
+        answer_id = when_dict.get("id", None)
 
         if when_value:
             when_values.append(when_value)
 
         for rule_value in when_values:
-            if isinstance(rule_value, int):
-                valid_rule = True
-            else:
+            valid_rule = False
+
+            if answer_id is None:
+                return errors
+
+            if answer_ids_with_parent_id[answer_id]["answer"].get("type") in (
+                "Radio",
+                "Checkbox",
+            ):
                 valid_rule = self.is_rule_value_valid(
-                    answer_ids_with_parent_id, rule_value
+                    answer_ids_with_parent_id, rule_value, answer_id
```

Comment:
Would be better to modify this to only pass the required answer (answer_ids_with_parent_id[answer_id]["answer"]) to the method.

Reply:
Passing only answer now, thanks!